feature-libs/quote/core/http-interceptors/quote-bad-request.handler.ts

```diff
@@ -0,0 +1,146 @@
+      ({ message }: ErrorModel) => {
+        this.handleIllegalArgumentIssues(message as string);
+      }
+    );
+
+    if (this.getCartValidationErrors(response).length > 0) {
+      this.handleCartValidationIssues();
+    }
+
+    if (this.getDomainErrors(response).length > 0) {
+      this.handleDomainErrors(this.quoteCartService);
+    }
+  }
+
+  protected getQuoteThresholdErrors(response: HttpErrorResponse): ErrorModel[] {
+    return (response.error?.errors ?? []).filter(
+      (error: ErrorModel) => error.type === 'QuoteUnderThresholdError'
+    );
+  }
+
+  protected getCartValidationErrors(response: HttpErrorResponse): ErrorModel[] {
+    return (response.error?.errors ?? []).filter(
+      (error: ErrorModel) => error.type === 'CartValidationError'
+    );
+  }
+
+  protected getDomainErrors(response: HttpErrorResponse): ErrorModel[] {
+    return (response.error?.errors ?? []).filter(
+      (error: ErrorModel) => error.type === 'DomainError'
+    );
+  }
+
+  protected getIllegalArgumentErrors(
+    response: HttpErrorResponse
+  ): ErrorModel[] {
+    return (response.error?.errors ?? []).filter(
+      (error: ErrorModel) => error.type === 'IllegalArgumentError'
+    );
+  }
+
+  protected handleQuoteThresholdErrors(message: string) {
+    const unmetTresholdMask = /does not meet the threshold\./;
+    const result = message.match(unmetTresholdMask);
+
+    if (result) {
+      this.globalMessageService.add(
+        {
+          key: 'quote.httpHandlers.threshold.underThresholdError',
+        },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+    }
+  }
+
+  protected handleCartValidationIssues() {
+    this.globalMessageService.add(
+      {
+        key: 'quote.httpHandlers.cartValidationIssue',
+      },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  }
+
+  protected handleDomainErrors(quoteCartService: QuoteCartService) {
+    quoteCartService
+      .isQuoteCartActive()
+      .pipe(take(1))
+      .subscribe((isActive) => {
+        if (isActive) {
+          this.globalMessageService.add(
+            {
+              key: 'quote.httpHandlers.quoteCartIssue',
+            },
+            GlobalMessageType.MSG_TYPE_ERROR
+          );
+        }
+      });
+  }
+
+  protected handleIllegalArgumentIssues(message: string) {
+    const discountMask = /Discount type is absolute/;
+    const discountRelated = message.match(discountMask);
+    const expirationMask = /Invalid quote expiration time/;
+    const expirationRelated = message.match(expirationMask);
+
+    if (discountRelated) {
+      this.globalMessageService.add(
+        {
+          key: 'quote.httpHandlers.absoluteDiscountIssue',
+        },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+    }
+
+    if (expirationRelated) {
+      this.globalMessageService.add(
+        {
+          key: 'quote.httpHandlers.expirationDateIssue',
+        },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+    }
+  }
+
+  getPriority(): Priority {
```

Comment:
This method is only used in the test. Please, remove it and the test

Reply:
@ChristophHi, is it required to add a comment to this open API?

Reply:
Needed and specified by abstract super method